sw/otbn/crypto/p256.s

```diff
@@ -1691,6 +1693,133 @@ p256_scalar_mult:
 
   ret
 
+/**
+ * Generate a nonzero random value in the scalar field.
+ *
+ * Returns t, a random value in the range [1,n-1].
+ *
+ * This follows the method in FIPS 186-4 sections B.4.2 and B.5.2 for
+ * generation of secret scalar values d and k. The computation is:
+ *   do {
+ *     c = RBG(256); // fetch 256b random value
+ *   } while (c >= n - 1)
+ *   return c + 1;
+ *
+ * This implementation handles the unmasked secret value, but the seed is
+ * pulled from RND so it cannot be re-run with the same seed. This method
+ * should not be used for keymgr-derived seeds! However, it masks the secret
+ * scalar before returning so that it can be safely handled e.g. in scalarmult.
+ *
+ * Flags: Flags have no meaning beyond the scope of this subroutine.
+ *
+ * @param[in]  w31:  all-zero
+ * @param[out] w20:  first share of secret scalar t
+ * @param[out] w21:  second share of secret scalar t
+ *
+ * clobbered registers: x2, x3, x20, w20, w21, w29
+ * clobbered flag groups: FG0
+ */
+p256_random_scalar:
+  /* Load the curve order n.
+     w29 <= dmem[p256_n] = n */
+  li        x2, 29
+  la        x3, p256_n
```

Comment:
This is mostly for my own understanding. When we load parameters like this, should we add them to the function params, or is this assumed to be part of the global context from an interface perspective?

Reply:
Personally I distinguish between:
- constants like `p256_n` which are stored in `.data` within this file, can't be overridden by other linked assembly files, and should always hold the same values
- input/output buffers which are usually overridden by linked assembly files and are written/read by Ibex
I treat the first category as "not function parameters" but rather part of the global context, and the second as parameters.

Reply:
sg, thanks for clarifying!